.werft/observability/monitoring-satellite.ts

```diff
@@ -1,125 +1,161 @@
 
-    deployGitpodServiceMonitors(kubeconfig)
-    checkReadiness(kubeconfig, namespace, checkNodeExporterStatus)
-}
+        this.deployGitpodServiceMonitors();
+        this.checkReadiness();
+    }
 
-async function checkReadiness(kubeconfig: string, namespace: string, checkNodeExporterStatus: boolean) {
-    // For some reason prometheus' statefulset always take quite some time to get created
-    // Therefore we wait a couple of seconds
-    exec(`sleep 30 && kubectl --kubeconfig ${kubeconfig} rollout status -n ${namespace} statefulset prometheus-k8s`, {slice: sliceName, async: true})
-    exec(`kubectl --kubeconfig ${kubeconfig} rollout status -n ${namespace} deployment grafana`, {slice: sliceName, async: true})
-    exec(`kubectl --kubeconfig ${kubeconfig} rollout status -n ${namespace} deployment kube-state-metrics`, {slice: sliceName, async: true})
-    exec(`kubectl --kubeconfig ${kubeconfig} rollout status -n ${namespace} deployment otel-collector`, {slice: sliceName, async: true})
-
-    // core-dev is just too unstable for node-exporter
-    // we don't guarantee that it will run at all
-    if(checkNodeExporterStatus) {
-        exec(`kubectl --kubeconfig ${kubeconfig} rollout status -n ${namespace} daemonset node-exporter`, {slice: sliceName, async: true})
+    private async checkReadiness() {
+        const { kubeconfigPath, satelliteNamespace } = this.options;
+
+        // For some reason prometheus' statefulset always take quite some time to get created
+        // Therefore we wait a couple of seconds
+        exec(
+            `sleep 30 && kubectl --kubeconfig ${kubeconfigPath} rollout status -n ${satelliteNamespace} statefulset prometheus-k8s`,
+            { slice: sliceName, async: true },
+        );
+        exec(`kubectl --kubeconfig ${kubeconfigPath} rollout status -n ${satelliteNamespace} deployment grafana`, {
+            slice: sliceName,
+            async: true,
+        });
+        exec(
+            `kubectl --kubeconfig ${kubeconfigPath} rollout status -n ${satelliteNamespace} deployment kube-state-metrics`,
+            { slice: sliceName, async: true },
+        );
+        exec(
+            `kubectl --kubeconfig ${kubeconfigPath} rollout status -n ${satelliteNamespace} deployment otel-collector`,
+            { slice: sliceName, async: true },
+        );
+
+        // core-dev is just too unstable for node-exporter
+        // we don't guarantee that it will run at all
+        if (this.options.withVM) {
+            exec(
+                `kubectl --kubeconfig ${kubeconfigPath} rollout status -n ${satelliteNamespace} daemonset node-exporter`,
+                { slice: sliceName, async: true },
+            );
+        }
     }
-}
 
-async function deployGitpodServiceMonitors(kubeconfig: string) {
-    const werft = getGlobalWerftInstance()
+    private async deployGitpodServiceMonitors() {
+        const { werft, kubeconfigPath } = this.options;
```

Comment:
back when I implemented this I had no idea what was this `async` and was just adding because it was being used everywhere.

Does it make sense in this class? Almost function is using async, although we're not using Promises anywhere. Would it make sense to just remove it?